### ISSUE TITLE
geo_context accepts objects that aren't described in OpenAPI schema (MAPCO-4775)

### ISSUE THREAD

**Issue by NivGreenstein** [MapColonies/geocoding#25]
**Describe the bug**
geo_context query parameter accepts random objects like `{ "idk": "nothing" }` or even simple `{}`

**To Reproduce**
1. run in dev mode (`npm run start:dev`)
2. send a query that contains geo_context as query parameter (for example `/search/control/tiles`)
```curl
curl --location --globoff 'localhost:8080/search/location/query?query=airport%2C%20log%20angeles&geo_context={}&geo_context_mode=filter' \
--header 'x-api-key: 1' \
--header 'x-user-id: 1'
```
**Expected behavior**
It should send 400 bad request and a clear message about geo_context not being sent as described in OpenAPI schema. 

**Screenshots**
![image](https://github.com/user-attachments/assets/be87b7e9-75ec-405b-acce-24a64ce905fd)

**Additional context**
Because Swagger OpenAPI doesn't officially support object that can pass via query parameters, we had to use anyOf instead oneOf. The use of oneOf in this situation is buggy. We should implement server-side validation. Recommended package: [Ajv JSON schema validation.](https://ajv.js.org/)

**Comment by NivGreenstein** [comment]
For `/search/control/` routes there is a validation function `validateGeoContext` under `/control/utils.ts`. Need to also implement it for `/search/location/query` route.
<img width="1299" alt="image" src="https://github.com/user-attachments/assets/c28ed93f-fb1c-4310-bfec-558c9dc10327">

**Comment by NivGreenstein** [comment]
fixed at the #29
